crates/anvil/tests/it/fork.rs

```diff
@@ -1465,6 +1465,36 @@ async fn test_reset_dev_account_nonce() {
     assert!(receipt.status());
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_set_erc20_balance() {
+    let config: NodeConfig = fork_config();
+    let address = config.genesis_accounts[0].address();
+    let (api, _) = spawn(config).await;
+    let info = api.anvil_node_info().await.unwrap();
+    let number = info.fork_config.fork_block_number.unwrap();
+    assert_eq!(number, BLOCK_NUMBER);
+
+    alloy_sol_types::sol! {
+       #[sol(rpc)]
+       contract ERC20 {
+            function balanceOf(address owner) public view returns (uint256);
+       }
+    }
+    let dai = address!("0x6B175474E89094C44Da98b954EedeAC495271d0F");
+    let alloy_provider = alloy_provider::ProviderBuilder::new()
+        .connect("https://reth-ethereum.ithaca.xyz/rpc")
```

Comment:
Please use `next_http_*` fn to rotate / pick up new endpoints as in https://github.com/foundry-rs/foundry/blob/daad0d19b3cd0f760ba140e40a22bb65d0a5a5ee/crates/test-utils/src/rpc.rs#L86